samples/minimaloboe/src/main/java/com/example/minimaloboe/MainActivity.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.minimaloboe
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.ViewModel
+import com.example.minimaloboe.ui.theme.SamplesTheme
+import kotlinx.coroutines.*
+
+class MainActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            SamplesTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colors.background
+                ) {
+                    MainControls()
+                }
+            }
+        }
+    }
+}
+
+class ExampleViewModel : ViewModel() {
+
+    var audioPlayer = AudioPlayer();
+
+    fun startAudio(): Int {
+        return audioPlayer.startAudio()
+
+    }
+
+    fun stopAudio(): Int {
+        return audioPlayer.stopAudio()
+    }
+}
+
+@Composable
+fun MainControls(viewModel: ExampleViewModel = ExampleViewModel()) {
+
+    // State that affects the UI.
+    var started by remember { mutableStateOf(false) }
+    var result by remember { mutableStateOf(1) }
+
+    Column {
+        Text(text = "Minimal Oboe!")
+        // START button
+        Button(
```

Comment:
nit: the button UI takes some time to run. Sound starts playing way before the UI is done updating. Do you know whether we can change the speed of the UI transitions so it takes less time?

Reply:
My buttons are fast on Pixel 7. Which device is slow?

Reply:
The UI takes about half a second for the full animation from purple -> gray -> black.

Reply:
This for Pixel 6.

Reply:
I tried on Oriole running TM and did not see slow animation.

Reply:
Let's tweak this later if still a problem.